### ISSUE TITLE
Ansible fails to use sftp to communicate to CoreOS remote

### ISSUE THREAD

**Issue by pcm32** [mitchellh/packer#3728]
Use case: I'm building an immutable image of CoreOS with kubernetes and other components. I'm testing this first using the VirtualBox builder. I'm basing work on a project that uses Ansible to provision the CoreOS machine.

When Ansible tries to upload files through sftp to the CoreOS machine for its initial bootstrapping, it fails after trying to initialise the sftp subsystem. CoreOS doesn't have a sftp-server binary, but instead uses sftp-internal. This sftp-internal runs inside sshd; pausing the build and connecting manually through sftp on the communicator port works, however ansible through packer comes up with a different port to connect to it (and through this new port, it fails). If the `sftp_command` is not set, it will try to invoke on the CoreOS machine `sftp-server -e` for which an executable doesn't exist (and produce an non zero exit for that part). If set to a dummy command that would execute without failure (like `echo 'sftp subsystem is already running'`) the process doesn't throw an error, but ansible fails to connect through sftp in the same way. The error produced by ansible is the same in both cases

```
==> virtualbox-iso: starting sftp subsystem
    virtualbox-iso: fatal: [default]: UNREACHABLE! => {"changed": false, "msg": "ERROR! SSH Error: data could not be sent to the remote host. Make sure this host can be reached over ssh", "unreachable": true}
```

It would be useful is Ansible through packer could use the existing (when it exists) sftp subsystem instead of trying to initialise a new one.

To replicate the error:
```
git clone https://github.com/pcm32/packer-coreos-test.git
cd packer-coreos-test/packer
packer build immutable_kubespray_coreos.json
```

Complete output of running is available on this [gist](https://gist.github.com/pcm32/0a58ef78fd6faada05c1b3c045499365).

Packer version: 0.10.1
Host: MacOS X 10.10.5

Any suggested workarounds for this case? Thanks!

**Comment by bhcleek** [comment]
Try using SCP. You can use the latest source or add `ANSIBLE_SCP_IF_SSH=True` to `ansible_env_vars`. Alternatively, you'll need to set `command` to a value that will run an sftp server in a container.

**Comment by pcm32** [comment]
Thanks @bhcleek . Last time I tried this, packer documentation said that it didn't support Ansible using scp but only sftp, will check if that works now.

**Comment by rickard-von-essen** [comment]
@pcm32 It supports SCP on master, not released yet. See #3861

**Comment by jrm16020** [comment]
@pcm32 I've not had much luck with this one but I'm using the Packer container at the moment.  Does this work for you?

`"sftp_command": "/usr/lib64/misc/sftp-server -e",`

@rickard-von-essen any chance Packer releases again soon?  The support for scp is one of a few changes that I've seen recently merged that I'd like to take advantage of.

**Comment by bhcleek** [comment]
@pcm32 have you tried using SCP?

**Comment by bhcleek** [comment]
@pcm32 have you tried using SCP by setting `ANSIBLE_SCP_IF_SSH=True`?

**Comment by pcm32** [comment]
Hi @bhcleek sorry, I haven't tried this again. In the end I used a different OS. Will report back if I try again with CoreOS. Thanks!

**Comment by rickard-von-essen** [comment]
@mwhooker are working hard to get a release out. Expect one in a week or so.

**Comment by jrm16020** [comment]
Thanks @rickard-von-essen 

@bhcleek @rickard-von-essen using SCP works for me on CoreOS using the unreleased Packer 0.11.  @pcm32 after using Packer 0.11 and [this role](https://github.com/defunctzombie/ansible-coreos-bootstrap) I was able to build my image.

**Comment by rickard-von-essen** [comment]
👍 
Fixed by #3861
